src/data_collection.py

```diff
@@ -228,106 +268,170 @@ def calculate_individual_metrics(json_file_name="raw_data_storage"):
+):
     """Create and print the table using prettytable.
 
-    Use individual_metrics_storage as default file unless otherwise chosen.
+    Unless specified be sending data_dict=DICT as a parameter, the function.
+
+    will print from the file.
     """
-    # Initialize pretty table object
+    # Default headings are mentioned above in the parameter
+    if not len(data_dict) == 0:
+        dictionary = data_dict
+    else:
+        dictionary = json_handler.get_dict_from_json_file(file_name)
+    # Initialize a PrettyTable instance
     data_table = PrettyTable()
-    # retreive data from the appropriate json file
-    dictionary = json_handler.get_dict_from_json_file(file_name)
-    # Add needed headings
-    headings = [
-        "Username",
-        "Email",
-        "Commits",
-        "+",
-        "-",
-        "Total",
-        "Modified Lines",
-        "Lines/Commit",
-        "File Types",
-    ]
-    data_table.field_names = headings
-    for key in dictionary:
-        data_table.add_row(
-            [
-                key,
-                dictionary[key]["EMAIL"],
-                dictionary[key]["COMMITS"],
-                dictionary[key]["ADDED"],
-                dictionary[key]["REMOVED"],
-                dictionary[key]["TOTAL"],
-                dictionary[key]["MODIFIED"],
-                dictionary[key]["RATIO"],
-                dictionary[key]["FORMAT"],
-            ]
-        )
+    # add the username as a category for teh headings
+    data_table.field_names = ["Username"] + headings
+    # Loop through every author in the dictionary
+    for author in dictionary:
+        # Add the authors name to the current row
+        current_row = [author]
+        for heading in headings:
+            # add the value from every heading to the current row
+            current_row.append(dictionary[author][heading])
+        # add row to prettytable instance
+        data_table.add_row(current_row)
+        # reset the current row to an empty list for the next iteration
+        current_row = []
     print(data_table)
 
 
-# NOTE: For the purposes of testing and demo
+# NOTE: not testable
+def find_repositories(repo):
+    """Locates a Github repository with the URL provided by the user."""
+    # ask the user for a URL of a Github repository
+    miner = RepositoryMining(path_to_repo=repo)
+    return miner
 
-# Steps to run data_collection.py and display the data table:
 
-# Must be in the repository folder
-# Run the python file using `pipenv run python src/data_collection.py`
-# Enter the name of the json file you want the data to be written to
-# If prompted, due to data not being collected previously, enter the URL
-# or path of the Repository you would like to analyze
-# If the current repository is the one you would like to analyze, simply
-# hit enter wihtout typing anything.
+def merge_metric_and_issue_dicts(metrics_dict, issues_dict):
+    """Receive two dicitionaries one for issues and the other for metrics.
```

Comment:
Might improve readability to push the docstring text underneath the three quotation marks.